web/modules/custom/dbcdk_community/src/Plugin/Block/ProfileBlock.php

```diff
@@ -75,14 +76,10 @@ public function build() {
     // exceptions and log them so the site can continue running and display an
     // empty table instead of a fatal error.
     $profile = NULL;
-    $username = $this->getContext('username')->getContextData()->getValue();
     try {
-      $filter = [
-        'where' => [
-          'username' => $username,
-        ],
-      ];
-      $profile = $this->profileApi->profileFindOne(json_encode($filter));
+      $profile = $this->profileRepository->getProfileByUsername(
+        $this->getContextValue('username')
+      );
```

Comment:
I believe you're right and it's not anything you should change - I just saw an irregularity from the way you normally write your code (I know you!!) :smile: